content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/get_retirement_entrypoint.rb

```diff
@@ -1,15 +1,49 @@
-service = $evm.root['service']
-if service.nil?
-  $evm.log('error', 'get_retirement_entrypoint: missing service object')
-  exit MIQ_ABORT
-end
+# frozen_string_literal: true
+
+#
+# Description: Resolves Service retirement entry point if present
+#
+
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module Methods
+            class GetRetirementEntryPoint
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                retirement_entrypoint
+              end
+
+              private
+
+              def retirement_entrypoint
+                service = @handle.root['service']
+                if service.nil?
+                  @handle.log('error', 'get_retirement_entrypoint: missing service object')
+                  exit MIQ_ABORT
+                end
```

Comment:
@GregP Let's move this into a separate method that either returns the `service` object or raises an error.

Reply:
@gmcculloug Done